packages/vite/src/node/server/index.ts

```diff
@@ -601,6 +601,11 @@ export async function _createServer(
       return environment.transformRequest(url)
     },
     warmupRequest(url, options) {
+      warnFutureDeprecation(
+        config,
+        'removeServerTransformRequest',
+        'server.warmupRequest() is deprecated. Use environment.warmupRequest() instead.',
+      )
```

Comment:
Similar to https://github.com/vitejs/vite/pull/20436#discussion_r2224250645, I think we need a new deprecation name like `'removeServerWarmupRequest'`